### PR TITLE
[move vm] Added Constant pool     

### PR DESCRIPTION
**Pull request by tnowacki**
## Motivation

- New pool for constants
- The constants can support any non-struct value, serialized in the same format as on chain values
- Addresses are split
  - Runtime constants are in the constant pool
  - Namespace constants are in the 'AddressIdentifierPool'

## Follow Up

- Actually fix various generation tools
- Test that structs are banned 

## Test Plan

- cargo test 
- new tests

### CODE REVIEW

**Comment by sblackshear** [comment]
> The constants can support any non-struct value, serialized in the same format as on chain values

Is (e.g.) `vector<vector<u64>>` allowed?

File: language/move-vm/runtime/src/interpreter.rs [hunk 1]
```diff
@@ -274,20 +274,22 @@ impl<'txn> Interpreter<'txn> {
...
                         )?;
-                        self.operand_stack.push(Value::vector_u8(v))?;
+                        let type_ = runtime.resolve_signature_token(
+                            frame.module(),
+                            &constant.type_,
+                            frame.ty_args(),
+                            context,
+                        )?;
+                        self.operand_stack
+                            .push(Value::simple_deserialize(&constant.data, type_)?)?
```
Comment (dariorussi) [inline language/move-vm/runtime/src/interpreter.rs:292]: ouch...

**Comment by dariorussi** [comment]
> > The constants can support any non-struct value, serialized in the same format as on chain values
> 
> Is (e.g.) `vector<vector<u64>>` allowed?

yes it is

**Pull request closed without merging**